tests/integration/src/test/java/org/mule/test/core/context/notification/processors/ModuleComponentPathTestCase.java

```diff
@@ -0,0 +1,310 @@
+                                   asList(new DefaultComponentLocation.DefaultLocationPart("flowWithSetPayloadTwoTimesTwice",
+                                                                                           FLOW_TYPED_COMPONENT_IDENTIFIER,
+                                                                                           CONFIG_FILE_NAME,
+                                                                                           of(36))));
+
+  private static final DefaultComponentLocation FLOW_WITH_PROXY_SET_PAYLOAD_HARDCODED =
+      new DefaultComponentLocation(of("flowWithProxySetPayloadHardcoded"),
+                                   asList(new DefaultComponentLocation.DefaultLocationPart("flowWithProxySetPayloadHardcoded",
+                                                                                           FLOW_TYPED_COMPONENT_IDENTIFIER,
+                                                                                           CONFIG_FILE_NAME,
+                                                                                           of(41))));
+
+  private static final Optional<TypedComponentIdentifier> MODULE_OPERATION_CHAIN =
+      of(builder().withIdentifier(buildFromStringRepresentation("mule:module-operation-chain"))
+          .withType(PROCESSOR).build());
+
+  private static final Optional<TypedComponentIdentifier> LOGGER =
+      of(builder().withIdentifier(buildFromStringRepresentation("mule:logger"))
+          .withType(PROCESSOR).build());
+  private static final Optional<TypedComponentIdentifier> SET_PAYLOAD =
+      of(builder().withIdentifier(buildFromStringRepresentation("mule:set-payload"))
+          .withType(PROCESSOR).build());
+
+  final ProcessorNotificationStore listener = new ProcessorNotificationStore();
+
+  @Override
+  protected void doSetUp() throws Exception {
+    super.doSetUp();
+    listener.setLogSingleNotification(true);
+    muleContext.getNotificationManager().addListener(listener);
+  }
+
+  @Test
+  public void flowWithSingleMp() throws Exception {
+    //simple test to be sure the macro expansion doesn't mess up the a flow that has no modifications
+    flowRunner("flowWithSingleMp").run();
+    assertNextProcessorLocationIs(FLOW_WITH_SINGLE_MP_LOCATION.appendLocationPart("processors", empty(), empty(), empty())
+        .appendLocationPart("0", LOGGER, CONFIG_FILE_NAME, of(16)));
+    assertNoNextProcessorNotification();
+  }
+
+  @Test
+  public void flowWithSetPayloadHardcoded() throws Exception {
+    flowRunner("flowWithSetPayloadHardcoded").run();
+    assertNextProcessorLocationIs(FLOW_WITH_SET_PAYLOAD_HARDCODED.appendLocationPart("processors", empty(), empty(), empty())
+        .appendLocationPart("0", MODULE_OPERATION_CHAIN, MODULE_SIMPLE_FILE_NAME, of(12)));
+    assertNextProcessorLocationIs(FLOW_WITH_SET_PAYLOAD_HARDCODED.appendLocationPart("processors", empty(), empty(), empty())
```

Comment:
This location should only be related to the module.xml file. The path would be something like: operationName/processors/0